pangeo_forge_recipes/serialization.py

```diff
@@ -0,0 +1,133 @@
+from collections.abc import Collection
+from dataclasses import asdict
+from enum import Enum
+from hashlib import sha256
+from json import dumps
+from typing import Any, List, Sequence
+
+from .patterns import FilePattern, Index
+
+
+def json_default(obj: Any):
+    """For objects which are not serializable with ``json.dumps``, this function defines
+    type-specific handlers which extract a serializable value from the object.
```

Comment:
This docstring feels a little inconsistent with the actual behavior. The function might just hash obj, which is not the same as serializing it. Maybe we should call the function `either_encode_or_hash`.

Reply:
👍 https://github.com/pangeo-forge/pangeo-forge-recipes/pull/349/commits/d3489f5b046eb87c66ae64c8ae02cb12c2f0d6b4